File: backend-py/req/get_settings.py
import requests
from dataclasses import dataclass
from typing import Optional
from config import STRAPI_URL

def get_settings(jwt: str) -> Optional[dict]:
    url = f"{STRAPI_URL}/einstellung"
    headers = {
        "Authorization": f"Bearer {jwt}"
    }
    response = requests.get(url, headers=headers)
    response.raise_for_status()  # Raise an exception for HTTP errors
    res = response.json()
    data = res['data']
    attributes = data['attributes']
    return attributes

File: backend-rust/req/get_settings.rs
use config::STRAPI_URL;
use serde::Deserialize;

#[derive(Deserialize, Debug)]
pub struct EinstellungenData {
    pub data: IdAtr,
}

#[derive(Deserialize, Debug)]
pub struct IdAtr {
    pub id: i16,
    pub attributes: Einstellungen,
}

#[derive(Deserialize, Debug)]
#[allow(non_snake_case)]
pub struct Einstellungen {
    pub Barcode_Mindestlaenge: i32,
    pub Leitcodes_Aktiv: bool,
    pub Ausnahmen_Aktiv: bool,
}

// get all exceptions from the database
#[tokio::main]
pub async fn get_settings(jwt: &str) -> Result<EinstellungenData, reqwest::Error> {
    let url = format!("{}/api/einstellung", STRAPI_URL);
    let client = reqwest::Client::new();

    let res = client
        .get(&url)
        .header("Authorization", format!("Bearer {}", jwt))
        .send()
        .await?
        .json::<EinstellungenData>()
        .await?;

    Ok(res)
}
